Sources/Mailgun/Models/TemplateMessage.swift

```diff
@@ -0,0 +1,89 @@
+import Vapor
+
+extension Mailgun {
+    public struct TemplateMessage: Content {
+        public static var defaultContentType: MediaType = MediaType.formData
+        
+        public typealias FullEmail = (email: String, name: String?)
+        
+        public let from: String
+        public let to: String
+        public let replyTo: String?
+        public let cc: String?
+        public let bcc: String?
+        public let subject: String
+        public let template: String
+        public let templateData: [String:String]?
+        public let templateVersion: String?
+        public let templateText: Bool?
+        public let attachment: [File]?
+        public let inline: [File]?
+        
+        private enum CodingKeys : String, CodingKey {
+            case from, to, replyTo = "h:Reply-To", cc, bcc, subject, template, templateData = "h:X-Mailgun-Variables", templateVersion = "t:version", templateText = "t:text", attachment, inline
+        }
+        
+        public func encode(to encoder: Encoder) throws {
+            var container = encoder.container(keyedBy: CodingKeys.self)
+            try container.encode(from, forKey: .from)
+            try container.encode(to, forKey: .to)
+            try container.encode(cc, forKey: .cc)
+            try container.encode(bcc, forKey: .bcc)            
+            try container.encode(subject, forKey: .subject)
+            try container.encode(template, forKey: .template)
+            let jsonData = try! JSONEncoder().encode(templateData)
+            let jsonString = String(data: jsonData, encoding: .utf8)!
+            try container.encode(jsonString, forKey: .templateData)
+            try container.encode(templateVersion, forKey: .templateVersion)
+            let text = templateText != nil && templateText! ? "yes" : nil // need to send yes as string
```

Comment:
looks like ` && templateText! ` is redundant
```swift
let text = templateText != nil ? "yes" : nil
```

Reply:
it needs to be not nil and true, does it work like this?

Reply:
Oh, I see, my bad, then like this
```swift
let text = templateText == true ? "yes" : nil
```